src/port_forwarding.rs

```diff
@@ -99,19 +106,19 @@ impl From<Box<Path>> for Socket<'static> {
 
 impl Socket<'_> {
     /// Create a new TcpSocket
```

Comment:
We can keep this function, having a function that accepts ToSocketAddrw can be useful.

Reply:
My possibly controversial take is that making it easy to call `ToSocketAddrs` on the client is likely to lead to more bugs like the one that led me to file this PR! When using SSH port forwarding to tunnel through an SSH bastion host—which I imagine is the common use case—it's extremely likely that the client cannot resolve the hostname, because the hostname is private to the network with the bastion host. ISTM like the API here should encourage deferring resolution of hostnames, so that `openssh` can perform the resolution on the server if appropriate.

Reply:
Got it, this sounds reasonable to me.